File: rules_ocaml/ffi/ctypes/libfoo2/cclib/foo.c
#include <stdio.h>
#include "foo.h"

int foo_init(void)
{
    printf("initializing libfoo\n");
    return 0;
}


int foo_fnubar(char *s)
{
    printf("libfoo: computing fnubar: %s\n", s);
    return 99;
}

void foo_exit(void)
{
    printf("exiting libfoo\n");
}

File: rules_ocaml/ffi/ctypes/libfoo2/foo.ml
let () =
  if (C.Types.foo_version <> 1) then
    failwith "foo only works with libfoo version 1";

  match C.Functions.foo_init () with
  | 0 ->
    C.Functions.foo_fnubar "fnubar!";
    C.Functions.foo_exit ()
  | err_code ->
    Printf.eprintf "foo_init failed: %d" err_code;
;;
